update tests for webpack 5 (#395)
GoogleChromeLabs/tooling.report: tests/code-splitting/subtests/between-new-worker-type/webpack/index.md

---
result: partial
---

It's possible to extend the integrated WebWorker support to new or custom Worker types

This makes it possible to bundle code for use in new worker-like environments like Worklets. Code splitting between the "host" and "child" sets of bundles is also supported.

[worker-loader]: https://github.com/webpack-contrib/worker-loader

> [worker-plugin]: https://github.com/GoogleChromeLabs/worker-plugin

Reviewer: I think we don't need `worker-plugin` in webpack5, so lets remove this link from here as well.

Reviewer 2: Also probably worth removing worker-loader.

---
result: partial
---

> It's possible to extend the integrated WebWorker support to new or custom Worker types.

Reviewer 2: ```suggestion
It's possible to extend webpack's integrated Web Worker support to new or custom Worker types, as long as `importScripts()` is supported.
```

> This makes it possible to bundle code for use in new worker-like environments like Worklets. Code splitting between the "host" and "child" sets of bundles is also supported.

Reviewer 2: I played around with this for a bit and the only issue I ran into was that the logic for loading bundles assumes a Classic Worker context with `importScripts()` (even if Webpack is configured to [output modules](https://webpack.js.org/configuration/output/#outputmodule)). Worklets and Module Workers both lack `importScripts()`, so I'd like to add a blurb mentioning that:

```suggestion
This makes it possible to bundle code for use in new worker-like environments like Worklets. Code splitting between the "host" and "child" sets of bundles is also supported. Currently, bundles are loaded using `importScripts()`, which means code splitting cannot be used with [Module Workers] or Worklets.

[module workers]: https://web.dev/module-workers
```

---
result: partial
---

It's possible to extend webpack's integrated Web Worker support to new or custom Worker types, as long as `importScripts()` is supported.

> This makes it possible to bundle code for use in new worker-like environments like Worklets. Code splitting between the "host" and "child" sets of bundles is also supported. Currently, bundles are loaded using `importScripts()`, which means code splitting cannot be used with [Module Workers] or Worklets.

Reviewer 3: This seems like it's saying one thing then the other. "You can use this for worklets", "You cannot use this for worklets".

Maybe just say that non-module workers are supported, but module workers and worklets are not. I'm kinda split on whether this counts as a 'partial', because new worker types seem to be module workers. WDYT @developit?

Reviewer 2: Ah yeah that's my bad - I think I added the bit about importScripts usage without realizing it read as a contradiction. We could just drop the "like Worklets" line?

```suggestion
This makes it possible to bundle code for use in new worker-like environments like Service Worker. Code splitting between the "host" and "child" sets of bundles is also supported. Currently, bundles are loaded using `importScripts()`, which means code splitting cannot be used with [Module Workers] or Worklets.
```

Reviewer 3: +1!
